### PR TITLE
Add .editorconfig for third-party editors.

### PR DESCRIPTION
**Pull request by N3X15**
# Executive Summary
This PR adds a file that automatically configures many third-party editors (with the correct plugin) with appropriate coding styles.

# Editor Support
[See here](http://editorconfig.org/#download)

This does not affect game code, just makes shit easier for developers and enforces a standard tab style.

### CODE REVIEW

**Comment by SysadminJeroen** [comment]
Let me just confirm the current tab style: One tab and 3 spaces, right?

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# http://editorconfig.org/
+# Atom: Install the editorconfig addon.
+root = True
+
+[*.dm,*.dme]
+indent_style = tab
+indent_size = 2
```
Comment (PJB3005) [inline .editorconfig:7]: Wait what.

Comment (SysadminJeroen) [reply .editorconfig:7]: You're right @PJB3005, it should be `indent_size = 4` or `indent_size = 7`.

Comment (Exxion) [reply .editorconfig:7]: >7

What fresh meme

Comment (PJB3005) [reply .editorconfig:7]: No.

Hotdogs.

Comment (SysadminJeroen) [reply .editorconfig:7]: @PJB3005 `indent_style = 🌭` and `indent_stize = 1`?

Comment (PJB3005) [reply .editorconfig:7]: Genius!

**Pull request merged**